backend/app/rag/indices/vector_search/filters/filter_evaluator.py

```diff
@@ -0,0 +1,277 @@
+        FilterOperator.ANY,  # array of strings
+        FilterOperator.ALL,  # array of strings
+        FilterOperator.CONTAINS,  # string or number
+    }
+    
+    # Define operators that support text operations
+    TEXT_OPERATORS = {
+        FilterOperator.TEXT_MATCH,  # string
+        FilterOperator.TEXT_MATCH_INSENSITIVE,  # string
+    }
+    
+    # Define operators that support basic comparisons (string, int, float)
+    BASIC_OPERATORS = {
+        FilterOperator.EQ,  # string, int, float
+        FilterOperator.NE,  # string, int, float
+    }
+    
+    @staticmethod
+    def _validate_numeric_operator(operator: FilterOperator, value: Any) -> bool:
+        """Validate the value type for numeric operators"""
+        if not isinstance(value, (int, float)):
+            logger.error(f"Operator {operator} requires numeric value (int or float), got {type(value)}")
+            return False
+        return True
+    
+    @staticmethod
+    def _validate_array_operator(operator: FilterOperator, value: Any) -> bool:
+        """Validate the value type for array operators"""
+        if operator in [FilterOperator.ANY, FilterOperator.ALL]:
+            # ANY and ALL operators require array of strings
+            if not isinstance(value, (list, tuple)):
+                logger.error(f"Operator {operator} requires array of strings, got {type(value)}")
+                return False
+            if not all(isinstance(item, str) for item in value):
+                logger.error(f"Operator {operator} requires array of strings, got array with non-string items")
+                return False
+        else:
+            # IN, NIN, CONTAINS operators accept string or number
+            if not isinstance(value, (list, tuple)):
+                logger.error(f"Operator {operator} requires array value, got {type(value)}")
+                return False
+            if not all(isinstance(item, (str, int, float)) for item in value):
+                logger.error(f"Operator {operator} requires array of strings or numbers, got array with invalid items")
+                return False
+        return True
+    
+    @staticmethod
+    def _validate_text_operator(operator: FilterOperator, value: Any) -> bool:
+        """Validate the value type for text operators"""
+        if not isinstance(value, str):
+            logger.error(f"Operator {operator} requires string value, got {type(value)}")
+            return False
+        return True
+    
+    @staticmethod
+    def _validate_basic_operator(operator: FilterOperator, value: Any) -> bool:
+        """Validate the value type for basic comparison operators"""
+        if not isinstance(value, (str, int, float)):
+            logger.error(f"Operator {operator} requires string, int, or float value, got {type(value)}")
+            return False
+        return True
+    
+    @staticmethod
+    def evaluate_filter(value: Any, filter_value: Any, operator: FilterOperator, 
```

Comment:
I prefer to split into two methods, e.g. `evaluate_sql_filter` and `evaluate_in_memory_filter`, which simplifies the branching logic.

Is it possible to move FilterEvaluator to a more common location, maybe backend/app/rag/filters.py? Maybe `KnowledgeGraphIndex` can also reuse this filter.